documentation/Advanced Features/Objects and Extensions.md

```diff
@@ -0,0 +1,269 @@
+# Objects and Extension Methods
+
+Vyxal is already turing complete, and can be used to easily solve many problems.
+However, say there was a situation where you needed to model a complex real-world
+object, such as a car. You could use two lists to map keys to values, but that
+would require making sure that the two lists are always in sync. Objects take
+away this hassle, and allow you to model real-world objects in a way that is
+convienient and easy to use.
+
+## The OOP Model
+
+Unlike most OOP languages, Vyxal does not have traditional classes. Instead, it
+has struct-like objects. These objects contain only data, and no methods. To add
+methods to an object, you can use extension methods (covered later).
+
+There is also no inheritance, meaning that composition is the only way to reuse
+code. This is not a bad thing, as it makes the implementation of objects much
+simpler.
+
+## Defining Objects
+
+To define an object:

```

Comment:
The terminology has been confusing me a bit. What you define with `#:O` is a struct, right? Not a prototype like JS? Because calling it an object implies it's a JS-like object that acts as a prototype for other objects, rather than a C-struct-like template for making objects

Reply:
You're right that it's more like a struct, but I wanted to avoid calling it that so as to not create confusion with the terminology used for control flow structures. Maybe there's a better term available that I don't know of, like record.

Reply:
Record sounds good to me. I think record might imply all members are public to some, but whatever